controller/src/main/java/cern/jarrace/controller/jvm/impl/SimpleAgentRunnerSpawner.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * © Copyright 2016 CERN. This software is distributed under the terms of the Apache License Version 2.0, copied
+ * verbatim in the file “COPYING”. In applying this licence, CERN does not waive the privileges and immunities granted
+ * to it by virtue of its status as an Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+package cern.jarrace.controller.jvm.impl;
+
+import cern.jarrace.commons.domain.Service;
+import cern.jarrace.controller.jvm.AgentRunnerSpawner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedInputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Implementation of {@link AgentRunnerSpawner} that uses an the {@link ProcessBuilder} class to start a new JVM
+ * running cern.jarrace.agent.AgentRunner#main.
+ *
+ * @author tiagomr
+ */
+public class SimpleAgentRunnerSpawner implements AgentRunnerSpawner {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SimpleAgentRegistrySpawner.class);
+    private static final String AGENT_RUNNER_MAIN_CASS = "cern.jarrace.agent.AgentRunner";
```

Comment:
Well it will then make controller depend on Agent... even if we do not need to redeploy.
IMHO we should have this AgentRunner class on the controller since its something we do not want to be changed on the fly (anyone can just  write their own class with the same fully qualified name and do whatever they want), so I STRONGLY advise on passing the ContainerRegistry, ContainerDiscoverer and AgentRunner to the controller package. Or leave the Agent package as it is making it private but expose the Agent interface into an other package that will be made public.